packages/@aws-cdk/pipelines/lib/synths/simple-synth-action.ts

```diff
@@ -300,6 +316,13 @@ export class SimpleSynthAction implements codepipeline.IAction {
       actionName: this.actionProperties.actionName,
       input: this.props.sourceArtifact,
       outputs: [this.props.cloudAssemblyArtifact, ...(this.props.additionalArtifacts ?? []).map(a => a.artifact)],
+
+      // Inclusion of the hash here will lead to the pipeline structure changing if the
+      // buildspec changes, and hence the pipeline being restarted. This is necessary if
+      // the users adds (for example) build or test commands to the buildspec.
```

Comment:
Not just the buildspec but anything around the CodeBuild project.

```suggestion
      // Inclusion of the hash here will lead to the pipeline structure changing for any
      // changes to the underlying CodeBuild project, and hence the pipeline being restarted.
      // This is necessary if the users adds (for example) build or test commands to the buildspec.
```